nav2_amcl/src/amcl_node.cpp

```diff
@@ -622,6 +630,27 @@ AmclNode::getInitialPoseStatusCallback(
   if(initial_pose_is_ready_) response->status = response->STATUS_OK;
 }
 
+void
+AmclNode::setAmclTfStateCallback(
+  const std::shared_ptr<cmr_msgs::srv::SetAmclTfState::Request> request,
+  std::shared_ptr<cmr_msgs::srv::SetAmclTfState::Response> response)
+{
+  is_tf_to_update_ = request->active;
+
+  if (!is_tf_to_update_) {
+    reset_tf_state_timer_ = this->create_wall_timer(
+      std::chrono::seconds(static_cast<int>(tf_state_reset_timeout_)),
+      [this] () -> void {
+        reset_tf_state_timer_->cancel();
```

Comment:
So when the timer is created, it is immediately cancelled?

Reply:
No, after `tf_state_reset_timeout_` seconds.